app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayController.kt

```diff
@@ -118,10 +127,10 @@ class DefaultTabsTrayController(
     private val navigationInteractor: NavigationInteractor,
     private val tabsUseCases: TabsUseCases,
     private val selectTabPosition: (Int, Boolean) -> Unit,
+    private val activity: HomeActivity,
```

Comment:
Let's reorder this to be the first param. Typically, we can derive a lot of these parameters from the activity since we will have access to the `context` and `context.components` and we can avoid long parameter list in further refactoring in the future.